research-software-directory/RSD-as-a-service
# 436 project overview

# Project overview page improvements

Closes #436 
Closes #456 

Changes proposed in this pull request:
* Label is improved to state "Updated .... ago"
* Items order implemented according to request: starting/running projects before finished ones and the newer projects first.
* Filter on keyword is added to project overview (similar to software keyword filter)
* Label improved on edit team member (project edit page)
* Refactoring: extending RPC's to with current_state property with values starting/running/finished based on project end date. Project card component updated to work with new values and related views on other pages like related projects on software page, projects on organisation page and project on personal page.

How to test:
* `docker-compose down --volumes` to remove old structure and rpc's
* `docker-compose build` to build new app
* `docker-compose up` to start
* optional: perform data migration to have something to work with
* confirm that project cards have label "Updated xxx ago" or Updated right now
* confirm that running projects are ordered before finished projects
* confirm that first project have started later (is newer) that the second project in the list
* navigate to software item and confirm that related projects are sorted in the same way
* navigate to project item and confirm that related projects are sorted in the same way
* navigate to organisation and select projects, confirm that card has also "Updated .... ago". The sorting should be similar, although maintainer van "pin" projects on the organisation wall, so the order is pinned, running, finished.
* navigate to personal page and confirm the project card has "Updated ... ago" and the order is similar

## Example 
 
![image](https://user-images.githubusercontent.com/9204081/181348065-295a9a19-da10-4f11-83ac-c78ed14e64f8.png)

## Team member fix
![image](https://user-images.githubusercontent.com/9204081/181443287-7c2ec64c-e13a-4faa-b478-e1088ce7f887.png)

PR Checklist:
